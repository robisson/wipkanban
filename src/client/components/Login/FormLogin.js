import React from 'react';
import {Redirect, Link} from 'react-router-dom'
import PropTypes from 'prop-types';

const FormLogin = ({
    state: {
        success,
        message
    },
    onLogin
}) => {
    let email;
    let password;
    
    if ((typeof window !== 'undefined') && window.localStorage.getItem('token')) {
        return <Redirect to="/"/>
    }

    return (
        <div className="container">
            <div
                className="row justify-content-md-center justify-content-sm-center align-items-center">
                <div className="col-xl-4 col-lg-4 col-md-6 col-sm-8">
                    <br/>
                    <div className="card">
                        <div className="card-header bg-murrey">
                            <h2 className="text-center text-white">WIPKanban</h2>
                        </div>
                        <p
                            className={success
                            ? 'alert-success'
                            : 'alert-danger'}>
                            {message}
                        </p>
                        <div className="card-body">
                            <form
                                onSubmit={(event) => onLogin(email.value, password.value, event)}
                                method="post"
                                action="/api/v1/login">
                                <div className="form-group">
                                    <label>
                                        <strong>Email:</strong>
                                    </label>
                                    <input
                                        required="required"
                                        placeholder="Type your email"
                                        className="form-control form-control-lg"
                                        type="email"
                                        ref={node => {
                                        email = node;
                                    }}/>
                                </div>
                                <div className="form-group">
                                    <label>
                                        <strong>Password:</strong>
                                    </label>
                                    <input
                                        required="required"
                                        placeholder="Type your password"
                                        className="form-control form-control-lg"
                                        type="password"
                                        ref={node => {
                                        password = node
                                    }}/>
                                    <a className="text-danger" href="#">I forgot my password</a><br/>
                                </div>
                                <div className="form-group text-right">
                                    <button className="btn btn-murrey btn-lg">
                                        Login
                                        <i className="fa fa-check"></i>
                                    </button>
                                </div>
                                <small>
                                    <Link to="/create-account" className="text-dark">
                                        <strong>No account yet? Create one now!</strong>
                                    </Link>
                                </small>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
};

FormLogin.propTypes = {
    state: PropTypes.object.isRequired,
    onLogin: PropTypes.func.isRequired
}

export default FormLogin;