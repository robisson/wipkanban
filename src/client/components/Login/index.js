// @flow
import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import InputAdornment from "@material-ui/core/InputAdornment";
import TextField from "@material-ui/core/TextField";
import LockIcon from "@material-ui/icons/Lock";
import EmailIcon from "@material-ui/icons/Email";
import Check from "@material-ui/icons/Check";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import { Redirect, Link } from "react-router-dom";
import classNames from "classnames";
import CircularProgress from "@material-ui/core/CircularProgress";
import green from "@material-ui/core/colors/green";

const styles = theme => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * 2,
    height: "100vh",
    display: "flex",
    alignItems: "center"
  },
  backgroundGreen: {
    backgroundColor: theme.palette.primary.main
  },
  text: {
    width: "100%"
  },
  link: {
    textDecoration: "none",
    color: theme.palette.secondary.main,
    "&:hover": {
      textDecoration: "underline"
    }
  },
  learnMore: {
    color: "white",
    borderColor: "white"
  },
  wrapper: {
    margin: theme.spacing.unit,
    position: "relative"
  },
  buttonProgress: {
    color: green[500],
    position: "absolute",
    top: "50%",
    right: 133.69 / 2 - 33 / 2,
    marginTop: -12,
    marginLeft: -12
  },
  buttonSuccess: {
    backgroundColor: green[500],
    "&:hover": {
      backgroundColor: green[700]
    }
  }
});

type Props = {
  classes: Object,
  state: Object,
  onLogin: Function,
  success: boolean,
  showPreloader: boolean
};

type State = {
  email: string,
  password: string,
  requiredFields: boolean
};

class Login extends React.Component<Props, State> {
  state: State = {
    email: "",
    password: "",
    requiredFields: false
  };

  _onLogin() {
    let { onLogin } = this.props;

    if (
      this.state.email.value.length == 0 ||
      this.state.password.value.length == 0
    ) {
      this.setState({ requiredFields: true });

      return;
    }

    onLogin(this.state.email, this.state.password);
  }

  render() {
    let {
      classes,
      state: { showPreloader, success }
    } = this.props;

    const buttonClassname = classNames({
      [classes.buttonSuccess]: success
    });

    if (typeof window !== "undefined" && window.localStorage.getItem("token")) {
      return <Redirect to="/" />;
    }

    return (
      <Grid container className={classes.root}>
        <Grid item xs={12} className={classes.demo}>
          <Grid
            container
            className={classes.demo}
            direction="row"
            justify="flex-start"
          >
            <Grid item xs={12} sm={12} md={7} lg={8} xl={8}>
              <Paper
                elevation={0}
                className={[classes.paper, classes.backgroundGreen].join(" ")}
                style={{
                  paddingTop: 10,
                  paddingBottom: 10
                }}
              >
                <div>
                  <br />
                  <Typography
                    style={{
                      color: "white"
                    }}
                    variant="h4"
                    gutterBottom
                  >
                    Welcome to the best open source plataform for lean
                    management
                  </Typography>
                  <Typography
                    style={{
                      color: "white"
                    }}
                    variant="h5"
                    gutterBottom
                  >
                    Align strategy and execution with <b>OKRs</b> and{" "}
                    <b>Kanban</b> for you, your team and your company.
                  </Typography>
                </div>
              </Paper>
            </Grid>
            <Grid item xs={12} sm={12} md={5} lg={4} xl={4}>
              <Paper
                elevation={0}
                className={classes.paper}
                style={{
                  paddingTop: 10,
                  paddingBottom: 10
                }}
              >
                <div style={{ margin: "0 auto" }}>
                  <Typography
                    variant="h4"
                    color="primary"
                    gutterBottom
                    align="center"
                  >
                    <b>WIPLean</b>
                  </Typography>
                  <div
                    style={{
                      width: "100%",
                      textAlign: "center"
                    }}
                  >
                    <TextField
                      disabled={showPreloader}
                      error={this.state.requiredFields}
                      inputRef={el => this.setState({ email: el })}
                      className={classes.text}
                      label="Type your email"
                      InputProps={{
                        endAdornment: (
                          <InputAdornment position="end">
                            <EmailIcon />
                          </InputAdornment>
                        )
                      }}
                    />
                    <TextField
                      disabled={showPreloader}
                      error={this.state.requiredFields}
                      inputRef={el => this.setState({ password: el })}
                      type="password"
                      className={classes.text}
                      label="Password"
                      InputProps={{
                        endAdornment: (
                          <InputAdornment position="end">
                            <LockIcon />
                          </InputAdornment>
                        )
                      }}
                    />
                    <div
                      style={{
                        textAlign: "right"
                      }}
                    >
                      <br />
                      <div className={classes.wrapper}>
                        <Button
                          onClick={this._onLogin.bind(this)}
                          size="large"
                          variant="contained"
                          color="primary"
                          className={buttonClassname}
                          disabled={showPreloader}
                        >
                          Sign In
                          <Check />
                        </Button>
                        {showPreloader && (
                          <CircularProgress
                            size={24}
                            className={classes.buttonProgress}
                          />
                        )}
                        <br />
                      </div>
                      <br />
                      <Typography variant="body1" gutterBottom align="right">
                        Do not have an account yet ?{" "}
                        <Link to="/create-account" className={classes.link}>
                          <b>Sign up!</b>
                        </Link>
                      </Typography>
                    </div>
                  </div>
                  <br />
                  <div>
                    <Typography variant="body1" gutterBottom>
                      Sign in with:
                    </Typography>
                    <br />
                    <IconButton className={classes.button}>
                      <i
                        className="fa fa-facebook-square fa-2x"
                        style={{
                          color: "#4267b2"
                        }}
                      />
                    </IconButton>
                    <IconButton className={classes.button}>
                      <i
                        style={{
                          color: "#db4437"
                        }}
                        className="fa fa-google-plus fa-2x"
                      />
                    </IconButton>
                    <IconButton className={classes.button}>
                      <i className="fa fa-github-square fa-2x" />
                    </IconButton>
                  </div>
                </div>
              </Paper>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default withStyles(styles)(Login);
