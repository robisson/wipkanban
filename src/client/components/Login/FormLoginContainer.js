import { connect } from "react-redux";
import * as actions from "../../actions/Login";
import FormLogin from "./FormLogin";

const mapDispatchToProps = dispatch => {
  return {
    onLogin(email, password, event) {
      dispatch(actions.login(email, password));
      event.preventDefault();
    }
  };
};

const mapStateToProps = state => ({ state: state.userReducer });

const FormLoginContainer = connect(mapStateToProps, mapDispatchToProps)(
  FormLogin
);

export default FormLoginContainer;
