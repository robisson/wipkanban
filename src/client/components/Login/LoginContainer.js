import { connect } from "react-redux";
import { login } from "../../actions/Login";
import FormLogin from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onLogin(email, password) {
      dispatch(login(email.value, password.value));
    }
  };
};

const mapStateToProps = state => ({ state: state.userReducer });

const FormLoginContainer = connect(mapStateToProps, mapDispatchToProps)(
  FormLogin
);

export default FormLoginContainer;
