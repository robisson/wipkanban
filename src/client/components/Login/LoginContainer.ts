import { connect } from "react-redux";
import { login, oauthFacebook } from "../../actions/Login";
import FormLogin from "./index";
import { Dispatch } from "redux";

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onLogin(email: string, password: string) {
      dispatch(login(email, password));
    },
    oauthFacebook(accessToken: string) {
      dispatch(oauthFacebook(accessToken));
    }
  };
};

const mapStateToProps = (state:any) => ({ state: state.userReducer });

const FormLoginContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(FormLogin);

export default FormLoginContainer;
