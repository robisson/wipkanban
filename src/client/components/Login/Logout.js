import React from 'react';
import * as actions from '../../actions/Login';
import {connect} from 'react-redux'

const Logout = ({logout}) => {
    return (
        <a className="text-white" href="#" onClick={logout}>Logout ></a>
    )
}

export default connect(undefined, {logout: actions.logout})(Logout);