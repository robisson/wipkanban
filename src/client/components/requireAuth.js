import React from "react";
import FormLoginContainer from "./Login/FormLoginContainer";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export default function(ComposedComponent) {
  const mapStateToProps = ({ userReducer: { success } }) => {

    return {
      authenticated: success
    };
  };

  const Authentication = props => {
    if (props.authenticated) {
      return <ComposedComponent {...props} />;
    }

    return <FormLoginContainer />;
  };

  Authentication.propTypes = {
    authenticated: PropTypes.bool.isRequired
  };

  return connect(mapStateToProps)(Authentication);
}
