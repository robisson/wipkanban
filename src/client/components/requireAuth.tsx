import React from "react";
import FormLoginContainer from "./Login/LoginContainer";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export default function(ComposedComponent: any) {
  const mapStateToProps = ({ userReducer: { success } }: any) => {
    return {
      authenticated: success
    };
  };

  const Authentication: React.StatelessComponent<any> = (props: any) => {
    if (props.authenticated) {
      return <ComposedComponent {...props} />;
    }

    return <FormLoginContainer />;
  };

  Authentication.propTypes = {
    authenticated: PropTypes.bool.isRequired
  };

  return connect(mapStateToProps)(Authentication);
}
