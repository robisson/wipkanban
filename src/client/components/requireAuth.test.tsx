import * as React from "react";
import requireAuth from "./requireAuth";
import configureStore from "../configureStore";
import { loginSuccess } from "../actions/Login";
import { createShallow } from "@material-ui/core/test-utils";
import { ShallowWrapper } from "enzyme";

describe("<requireAuth /> Component", () => {
  let wrapper: ShallowWrapper;
  let store: any;
  let shallow: any;
  let ComponentClass: any;

  beforeEach(() => {
    shallow = createShallow();
    store = configureStore();
  });

  it("should render the exactly component passed when user is authenticated", () => {
    store.dispatch(loginSuccess(true, null, {}));
    let AuthenticatedComponent = () => <div>Hello</div>;

    ComponentClass = requireAuth(<AuthenticatedComponent />);

    wrapper = shallow(<ComponentClass />);
    expect((wrapper.props() as any).auth).toEqual(undefined);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the Login component when user is not authenticated", () => {
    let AuthenticatedComponent = () => <div>Hello</div>;

    ComponentClass = requireAuth(<AuthenticatedComponent />);
    wrapper = shallow(<ComponentClass />);

    expect((wrapper.props() as any).auth).toEqual(undefined);
    expect(wrapper).toMatchSnapshot();
  });
});
