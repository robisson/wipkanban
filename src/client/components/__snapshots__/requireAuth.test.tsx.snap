// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<requireAuth /> Component should render the Login component when user is not authenticated 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<requireAuth /> Component should render the exactly component passed when user is authenticated 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
